commons/all/src/main/java/org/infinispan/commons/io/FileWatcher.java

```diff
@@ -0,0 +1,116 @@
+package org.infinispan.commons.io;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+import org.infinispan.commons.logging.Log;
+import org.infinispan.commons.logging.LogFactory;
+
+/**
+ * @since 15.0
+ */
+public class FileWatcher implements Runnable {
```

Comment:
Do we want to use a WatchService instead? https://docs.oracle.com/javase/tutorial/essential/io/notification.html It should theoretically be more lightweight (don't have to check all registered files) and more responsive when file system supports (no sleeps required).

Reply:
The initial implementation did use that, but read https://github.com/infinispan/infinispan/pull/12359#discussion_r1595566386

Reply:
Ah, yes. Sorry, I only looked at the code not the prior comments :D